components/node-labeler/cmd/run.go

```diff
@@ -84,13 +90,8 @@ var runCmd = &cobra.Command{
 			log.WithError(err).Fatal("unable to start node-labeler")
 		}
 
-		kClient, err := client.New(ctrl.GetConfigOrDie(), client.Options{})
-		if err != nil {
-			log.WithError(err).Fatal("unable to create client")
-		}
-
 		r := &PodReconciler{
-			kClient,
+			mgr.GetClient(),
```

Comment:
I am guessing the intent here is to leverage cache for read, rather than hit the API server.

What makes this change necessary for this PR? I ask because using cache means if cache becomes stale, we could have problems. For example, stale data could mean we don't add or remove taints when we should.

Unless we were having a perf issue, I would not include in this PR.

Reply:
I missed that sync time was 2m. I think that is okay in hindsight. I was worried because default sync time is large.